tests/system/dcos_service_marathon_tests.py

```diff
@@ -26,19 +28,25 @@ def test_framework_readiness_time_check():
     """Tests that an app is being in deployment until the readiness check is done."""
 
     fw = apps.fake_framework()
-    readiness_time = 30
+    readiness_time = 15
     fw['readinessChecks'][0]['intervalSeconds'] = readiness_time
 
     client = marathon.create_client()
     deployment_id = client.add_app(fw)
 
-    time.sleep(readiness_time - 10)  # not yet.. still deploying
-    deployment = client.get_deployment(deployment_id)
-    assert deployment['currentActions'][0]['readinessCheckResults'][0]['ready'] is False, \
-        "The application is read"
+    @retrying.retry(wait_fixed=1000, stop_max_attempt_number=30, retry_on_exception=common.ignore_exception)
+    def assert_in_deployment(deployment_id):
+        deployment = client.get_deployment(deployment_id)
+        assert deployment['currentActions'][0]['readinessCheckResults'][0]['ready'] is False, \
+            "The application is ready"
```

Comment:
nit: `s/The application is ready/Application's readiness check is green where it should still be red` or smth. along those lines.